marklogic-data-hub/src/main/java/com/marklogic/hub/util/MlcpRunner.java

```diff
@@ -73,26 +77,40 @@ public String getJobId() {
     public void run() {
         HubConfig hubConfig = getHubConfig();
 
-        Job job = Job.withFlow(flow)
-            .withJobId(jobId);
-        jobManager.saveJob(job);
+        Job job = null;
+        if (flow != null) {
+            job = Job.withFlow(flow)
+                    .withJobId(jobId);
+            jobManager.saveJob(job);
+        }
 
         try {
             MlcpBean bean = new ObjectMapper().readerFor(MlcpBean.class).readValue(mlcpOptions);
-            bean.setHost(databaseClient.getHost());
-            bean.setPort(databaseClient.getPort());
+            if (databaseClient != null) {
+                bean.setHost(databaseClient.getHost());
+                bean.setPort(databaseClient.getPort());
+            }
             if (database != null) {
                 bean.setDatabase(database);
             }
 
-            // Assume that the HTTP credentials will work for mlcp
-            bean.setUsername(hubConfig.getAppConfig().getAppServicesUsername());
-            bean.setPassword(hubConfig.getAppConfig().getAppServicesPassword());
 
-            File file = new File(mlcpOptions.get("input_file_path").asText());
-            String canonicalPath = file.getCanonicalPath();
-            bean.setInput_file_path(canonicalPath);
-            bean.setTransform_param("\"" + bean.getTransform_param() + ",job-id=" + jobId + "\"");
+            if (!"copy".equals(bean.getCommand().toLowerCase())) {
```

Comment:
MlcpRunner was used to run legacy 4.x input flows. Are these changes meant to get the test running ?

Reply:
I agree - I believe the only thing needed is the privilege addition so that MLCP from the command line will work. MlcpRunner is not in scope.

Reply:
The change is for a test to verify MLCP copy works with the operator role. The easiest way to test that was to leverage MlcpRunner.